php/libraries/User.class.inc

```diff
@@ -76,7 +76,7 @@ class User extends UserPermissions
         );
         $user_cid            = array();
         foreach ($user_centerID_query as $key=>$val) {
-            $user_cid[$key] =$val['CenterID'];
+            $user_cid[$key] = intval($val['CenterID']) //Cast values to int;
```

Comment:
Pedantic note but technically this isn't the same as casting so I changed the comment.

More importantly you commented out you semi-colon here so this isn't gonna work!

```suggestion
            // Treat string representation of ID as an int
            $user_cid[$key] = intval($val['CenterID']);
```

Reply:
Sure. I updated my suggestion so you can commit it more easily.

Reply:
Committed